src/main/kotlin/com/milaboratory/mixcr/cli/ValidationUtil.kt

```diff
@@ -23,3 +25,27 @@ fun ValidationException.Companion.chainsExist(chains: Chains?, usedGenes: Collec
         }
     }
 }
+
+fun ValidationException.Companion.requireKnownSpecies(species: String?, vararg additional: String?) {
+    for (s in (listOf(species) + additional)) {
+        val known = try {
+            VDJCLibraryRegistry.getDefaultLibrary(s)
+            true
+        } catch (ignore: Throwable) {
+            false
+        }
+        require(known) { "Unknown species: $s" }
+    }
+}
+
+fun ValidationException.Companion.requireKnownChains(chains: String?, vararg additional: String?) {
+    for (s in (listOf(chains) + additional)) {
+        require(s != null && Chains.parse(s) != null) { "Unknown chains: $s" }
+    }
+}
+
+fun ValidationException.Companion.requireKnownGeneFeature(gf: String?, vararg additional: String?) {
+    for (s in (listOf(gf) + additional)) {
+        require(s != null && GeneFeature.parse(s) != null) { "Unknown gene feature: $s" }
+    }
+}
```

Comment:
please, config ide to add new line at the end of the file automaticaly

Reply:
BTW --- do we have codestyle commited in the composite repo?

Reply:
yes, but this setting is not there